### ISSUE TITLE
Validate or restructure admin-supplied content fields for accessibility

### ISSUE THREAD

**Issue by marcysutton** [forem/forem#11842]
**Is your feature request related to a problem? Please describe.**

The sponsorship widget in the left sidebar of Forem has accessibility issues, including missing alt text for an image wrapped in a link–causing an empty anchor to be rendered. This content area and other similar textfields should be updated to either validate the freeform content for accessibility, or be restructured into separate data fields to ensure accessible markup.

**Describe the solution you'd like**

There are a few options:
- A [plain textarea](https://github.com/forem/forem/blob/ada2d0c7280c40fa2714165de07e23d4123f4332/app/views/admin/display_ads/_form.html.erb#L7) could have an accessibility checker run on it, for example the [axe linter](https://github.com/marketplace/axe-linter) or [CKEditor](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_a11y.html) in the case of a WYSIWYG.
- The textarea could instead be converted into structured data fields such as a link, image, and image alt text, preventing inaccessible markup from being inserted. It would be less flexible markup-wise, but easier to implement and enforce accessibility.

**Describe alternatives you've considered**

Fixing the accessibility issues themselves in Forem would satisfy these particular accessibility issues short-term, but wouldn't prevent them from occurring again on DEV or other Forem sites.

**Additional context**

Here are some screenshots from the accessibility issues found in the [axe extension](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US):

![axe extension showing a "Links need discernible text" issue](https://user-images.githubusercontent.com/1045233/101718162-843aa880-3a55-11eb-8564-41fb5aaf9884.png)

![The DEV homepage with the Chrome DevTools open to an image with no alt attribute](https://user-images.githubusercontent.com/1045233/101718209-9ae0ff80-3a55-11eb-9dd1-2bb52e84c0c1.png)

**Comment by marcysutton** [comment]
Here's another example that appeared recently, the "campaignmode-hackathon", which has images with no alt text and an h1 with only a `+` in it:

![Dev.to with a DigitalOcean Hackathon banner open in axe devtools](https://user-images.githubusercontent.com/1045233/101811290-c522d380-3ace-11eb-9b90-b49e27653004.png)
![The hackathon banner open in Chrome DevTools showing images with no alt text](https://user-images.githubusercontent.com/1045233/101811309-cc49e180-3ace-11eb-80b5-edd18012b324.png)

**Comment by cmgorton** [comment]
@forem/frontend I know we have had several accessibility changes recently. is this something we have worked on? If not I think we should add this either to our planning funnel or I can add this to the MLH project board. Thoughts?

**Comment by nickytonline** [comment]
This hasn't been worked on and most likely will require an RFC.

**Comment by mikeydorje** [comment]
Closing as stale. No activity since 2021.

If this issue persists on current DEV or your instance, please reopen with updated details/reproduction steps.
